_posts/2016-09-12-Microservices-Dashboard-1.0.1.md

```diff
@@ -0,0 +1,197 @@
+The Microservices Dashboard is a brand new open source project, which officially launched its first major release at Spring One Platform.
+Building on top of Spring Boot and Spring Cloud, it visualises your microservice architecture and integrates with tools every microservice architecture benefits from.
+This ranges from consumer-driven-contract testing over service discovery to hypermedia traversal and more.
+
+Microservices Dashboard is a simple application to visualize links between microservices and the encompassing ecosystem.
+This AngularJS application consumes endpoints exposed by its server component.
+It displays four columns: UI, Resources, Microservices and Backends.
+Each of these columns show nodes and links between them.
+The information for these links come from Spring Boot Actuator health endpoints, Pact consumer-driven-contract tests and hypermedia indexes, which are aggregated in the microservices-dashboard-server project.
+
+## The architecture
+
+<img class="center-block" alt="Architecture" src="/img/microservices-dashboard/architecture.png" width="50%">
+
+The dashboard currently consists out of an AngularJS 1.x application which speaks over HTTP to a Spring Boot application.
+The frontend uses D3.js to visualise the nodes in the four columns.
+We are currently in the process of completely rebuilding the frontend stack.
+Next version will be running on Angular 2, Typescript and EcmaScript 6.
+Most of D3.js will be taken care of by Angular 2 itself.
+
+Thanks to this refactor we’ll see the introduction of RxJS, making the frontend much more reactive in nature.
+This aligns our frontend and backend components goals, since the backend is already running RxJava.
+Our efforts currently focus on replicating all functionality which is currently available in the dashboard, albeit with much more attention to quality and testing.
+Subsequently new features and enhancements will be built on top of a much more mature and extendible frontend application.
+
+
+Our server component is powered by Spring Boot’s auto configuration.
+It’s a library which, once on a regular Spring Boot’s classpath, will automatically transform the Spring Boot application into a JSON graph-serving engine.
+It does so by using aforementioned RxJava.
+The idea of the server application is that it will fetch information from the microservices ecosystem, with for instance Spring Cloud's integration of service registries, and collect details of components and their relation within said ecosystem.
+Needless to say collecting this information requires a lot of outbound calls, and can pose a serious performance burden in case the landscape gets bigger.
+Making intelligent use of the system’s resources is absolutely necessary, and RxJava does just that.
+In the future we might migrate to Spring’s Reactor which has a more formal integration of the ReactiveX specification and better integration with Spring itself.
```

Comment:
I would make _Spring's Reactor_ clickable, directing the reader to more information about the new project.